cluster-up/cluster/kind-k8s-sriov-1.17.0/config_sriov.sh

```diff
@@ -12,6 +12,134 @@ WORKER_NODE_ROOT="${CLUSTER_NAME}-worker"
 
 OPERATOR_GIT_HASH=8d3c30de8ec5a9a0c9eeb84ea0aa16ba2395cd68  # release-4.4
 
+# This function gets a command string and invoke it
+# until the command returns an empty string or until timeout
+function retry {
+  local -r tries=$1
+  local -r wait_time=$2
+  local -r action=$3
+  local -r wait_message=$4
+
+  local result=$(eval $action)
+  for i in $(seq $tries); do
+    if [[ -z $result ]] ; then
+      echo "[$i/$tries] $wait_message"
+      sleep $wait_time
+      result=$(eval $action)
+    else
+      return 0
+    fi
+  done
+
+  return 1
+}
+
+function wait_for_daemonSet {
+  local name=$1
+  local namespace=$2
+  local required_replicas=$3
+
+  if [[ $namespace != "" ]];then
+    namespace="-n $namespace"
+  fi
+
+  if (( required_replicas < 0 )); then
+      echo "DaemonSet $name ready replicas number is not valid: $required_replicas" && return 1
+  fi
+
+  local -r tries=10
+  local -r wait_time=3
+  wait_message="Waiting for DaemonSet $name to have $required_replicas ready replicas"
+  error_message="DaemonSet $name did not have $required_replicas ready replicas"
+  action="_kubectl get daemonset $namespace $name -o jsonpath='{.status.numberReady}' | grep $required_replicas"
+
+  retry "$tries" "$wait_time" "$action" "$wait_message" && return  0
+  echo $error_message && return 1
+}
+
+function wait_pod {
+  local namespace=$1
+  local label=$2
+
+  local -r tries=60
+  local -r wait_time=5
+
+  local -r wait_message="Waiting for pods with $label to create"
+  local -r error_message="Pods with  label $label at $namespace namespace found"
+
+  if [[ $namespace != "" ]];then
+    namespace="-n $namespace"
+  fi
+
+  if [[ $label != "" ]];then
+    label="-l $label"
+  fi
+
+  local -r action="_kubectl get pod $namespace $label -o custom-columns=NAME:.metadata.name --no-headers"
+
+  retry "$tries" "$wait_time" "$action" "$wait_message" && return  0
+  echo $error_message && return 1
+}
+
+function wait_k8s_object {
+  local -r object_type=$1
+  local -r name=$2
+  local namespace=$3
+
+  local -r tries=60
+  local -r wait_time=3
+
+  local -r wait_message="Waiting for $object_type $name"
+  local -r error_message="$object_type $name at $namespace namespace found"
+
+  if [[ $namespace != "" ]];then
+    namespace="-n $namespace"
+  fi
+
+  local -r action="_kubectl get $object_type $name $namespace -o custom-columns=NAME:.metadata.name --no-headers"
+
+  retry "$tries" "$wait_time" "$action" "$wait_message" && return 0
+  echo $error_message && return  1
+}
+
+
+function is_taint_absence {
+  local -r taint=$1
+
+  result=$(_kubectl get nodes -o jsonpath="{.items[*].spec.taints[?(@.effect == \"$taint\")].effect}")
+  if [[ -z $result ]]; then 
+    echo "not-presnt"
+  fi
```

Comment:
dont you need else echo $result ?

Reply:
`is_taint_absence`should return empty string if the taint is presented,
If that case `$result=<taint effect name>`

Reply:
The comment was about the previous situation. since then the logic was adapted.
lets look please on the new one.

if result is empty its of course not "error", this means i would double check the condition to see if thats what you actually meant please. even that it works, the conditions arent ortogonal in their current AND semantic.

i checked by heart that it works on those 3 cases. hope we both arent wrong, 
if the condition above will do change, we need to recheck of course.

1. taint present - we need to retry because we want to wait taints to be removed
2. error due to hiccup - retry
3. no taint present and no error - we are done

Thanks

Reply:
Got it, I see what is the problem now thanks.
I pushed new commit to fix it.

Reply:
> dont you need else echo $result ?
I see the problem not, thanks for raising that and for previous comments
I pushed new commit and changed how it works

I checked all those cases on a cluster and it works as we expect :)

Reply:
great
i meant mostly that
`if [[ -z $result ]] && [[ $result != error ]];`
even if it works, some stuff might be redudent there, 
because if result is empty its of course not error (which should be "error", again even if it works)
food for thought

Reply:
In case of a hiccup or kubectl error `kubectl get nodes -o jsonpath..` return an error with error return code
casing the called function `wait_for_taint_absence` to terminate immediately.
With adding  `|| error` we handle that error by setting `result="error"` (with `ehco error`, could be any string)
thus that error wont terminate the caller function and `wait_for_taint_absence` will keep running until timeout.

Thanks for the feedback much appreciated ;)

Reply:
this i understood, its not what i meant, i mean only about the condition itself where the right part (comparing to error string), might be redundant.
i wont stall about it, just food for thought

Reply:
I checked again and you were right, the second literal ` [[ $result != error ]];` is redundant, I will remove it
Thanks.